### PR TITLE
onildac-inputclass

### PR DESCRIPTION
**Pull request by NathanSegerlind**
Created a case class for the oni-lda-c input so it would not rely on (the error prone, ugly) method of packing data into comma delimited strings

### CODE REVIEW

File: src/main/scala/org/opennetworkinsight/OniLDACWrapper.scala [hunk 1]
```diff
@@ -17,24 +17,26 @@ import scala.sys.process._
...
+
+  def runLDA(docWordCount: RDD[OniLDACInput], modelFile: String, topicDocumentFile: String, topicWordFile: String,
              mpiPreparationCmd: String, mpiCmd: String, mpiProcessCount: String, mpiTopicCount: String,
              localPath: String, ldaPath: String, localUser: String, dataSource: String, nodes: String):
   (Array[String], Array[String])
   =
   {
-    val documentWordData = docWordCount.map(_.split(","))
     // Create word Map Word,Index for further usage
     val wordDictionary: Map[String, Int] = {
```
Comment (rabarona) [inline src/main/scala/org/opennetworkinsight/OniLDACWrapper.scala:30]: In function createModel(...) _wordDictionary_ has a different name, _wordToIndex_, which I think is a good idea. Can we change _wordDictionary_ all over the code to keep the same definition of what this variable is and call it _wordWithIndex_?

**Comment by rabarona** [comment]
LGTM

File: src/main/scala/org/opennetworkinsight/OniLDACWrapper.scala (hunk 1 above)
Comment (NathanSegerlind) [inline src/main/scala/org/opennetworkinsight/OniLDACWrapper.scala:30]: yeah, just make a note of it and we can get it another pass ...

a lot of the code needs some variable rename and style passes... those should be easy runs but let's not forget anything

**Comment by NathanSegerlind** [comment]
just a quick explanatory note:

I had considered the possibility of having the oni-lda-c take a DataFrame,   but that seemed to be the wrong approach to me for a couple reasons
- big advantage of dataframes is the clean add columns syntax/functionality... but the oni-lda-c wrapper is only consuming the data, not adding new features to the input data set
- dataframe would require enforcing that the caller created a data frame with a schema that extends a provided schema (or having the caller specify three  columns as doc column, word column and count column), which, with validation, seemed to me more bother than just creating objects of a provided case class

**Pull request merged**